doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,587 @@
+            + `application` - Application for which the configuration is meant
+            + `scope` - Application internal scope
+            + `slug` - Entry specific unique slug
+    + filter (string, optional) - Only entries containing the search string in either their path, title, description, author or tags will be added to the output.
+    + filterby (enum[string], optional) - On what fields the filter should be applied to (i.e. what fields to involve in search)
+        + Default: `all`
+        + Members
+            + `all` - All fields: key, title, description, author and tags
+            + `key` - Only the key
+            + `title` - Only the title
+            + `description` - Only the description
+            + `author` - Only the authors name
+            + `tags` - Only tags
+
++ Response 200 (application/json)
+    + Attributes (Entry List)
+
+
+## Get details for entry [GET /database/{organization}/{application}/{scope}/{slug}{?raw}]
+
+Returns detailed information about the entry with the specified key (path). If the optional parameter `raw` is supplied, a raw configuration is returned instead of the normal response.
+
++ Parameters
+    + organization (string, required) - The `organization` who built the `application` for which the configuration of the entry is meant
+    + application (string, required) - The `application` to which the configuration of the entry belongs to
+    + scope (string, required) - The `application` specific `scope` to which the configuration of the entry belongs to
+    + slug (string, required) - The unique identifier of the requested entry
+    + raw (string, optional) - If this parameter is supplied, the API will attempt to return the requested configuration snippet in the given `raw` format
+
++ Response 200 (application/json)
+    + Attributes (object)
+        + data (object) - The actual details about the entry
+            + key (object) - The key and its separate parts
+                + full (string) - The full key (path) of the entry
+                + organization (string) - The organization of the application to which the snippet belongs to
+                + application (string) - The application to which the snippet belongs to
+                + scope (string) - The application internal scope for the configuration snippet
+                + slug (string) - The unique identification slug of the entry
+            + meta (object) - Additional information about the entry
+                + title (string) - The descriptive title of the entry
+                + description (string) - The extended description of the entry
+                + author (string) - The name of the creator of the entry
+                + created_at (number) - A timestamp of the creation date of the entry
+            + value (array[object])
+                + format (string) - A configuration format, e.g. **ini**
```

Comment:
The way I design my API also depends on these decisions, so I cannot always discuss it later. Every additional discussion takes time from me that I'm not ready to invest anymore. You know what goal I have in mind and I already see it vanishing - again.

You provided me with a title for my thesis, but not with concrete ideas or demands what the whole thing should be capable of. A title gives latitude to make own decisions and concepts. If we are now changing everything after I already made my plans, I'm not ok with it. Especially not if it takes dozens of iterations until you are satisfied or run out of ideas for new stuff. It prevents me from going on in the milestone list, I cannot implement stuff if it is likely to change again. I'm not willing to do all work two or even more times. Especially not in a programming language that I'm not using in daily business.

It is also hard for me to plan everything to the end beforehand, without writing a single line of code, as I don't know the whole library and all of its tools yet. I certainly will build my decisions upon capability and feasibility of the tools I've on my hand, even if this requires to make compromises concerning erstwhile plans and imaginations. Writing a second thesis about the recognition of configuration formats is not within the scope of my work and will never be.this format (and no meta information). It basically pipes the output of the storage export to the HTTP response. This way you could simply pipe the output of the HTTP call to a config file.

---

I made the decision to remove the guessing of the format completely as the current plugins are not stable enough to do it for me. Instead I'll offer to choose the format during upload. This will probably require a method to retrieve supported formats, but I'm fine with this. I don't want to offer override methods for this because of bad usability (everyone will first try the auto-detection only to be disappointed afterwards). 

---

As I'm no curl guru and I don't want to read up on this right now, I'll just add my existing use-cases to the PR. 

---

Finally, Blueprints are nice because they are easy to read. I understand that some things might seem weird, but the two operations in question are really easy to understand.

[The uploading of snippets](https://github.com/Namoshek/libelektra/blob/1d2c4b92405adf6e7555c2b751252f59bccac43b/doc/rest_api/snippet_sharing/api-description.apib#L370) is pretty straight forward. Within a JSON object you supply some meta information and a configuration as plain string (if this comes from curl, a textbox or a file doesn't matter at all - it is a plain string). We are then - somehow - finding the right storage plugin to use for the snippet, convert the whole snippet into a KeySet and store it with KDB. No magic happens here.

[The retrieval of snippets](https://github.com/Namoshek/libelektra/blob/1d2c4b92405adf6e7555c2b751252f59bccac43b/doc/rest_api/snippet_sharing/api-description.apib#L320) is even easier.
When the path to a snippet has been specified, a normal JSON response (the first 200 response in the list) will be returned, with the snippet already converted into all formats that the service supports.
If the additional `raw` parameter has been supplied (and contains a supported format), the converted snippet will be returned as plain text (the second 200 response). This response can then be piped into files or whatever suits the user best.